### DISCUSSION TITLE
How to disable the default routing and specify manual one in Nuxt 3?

### DISCUSSION THREAD

**Discussion by TokugawaTakeshi** [nuxt/nuxt#18794] [Questions] [answered]
I am asking this question after [has not been answered on Stack Overflow](https://stackoverflow.com/questions/75115935/how-to-disable-the-default-routing-and-specify-manual-one-in-nuxt-3) even after [started the bounty](https://stackoverflow.com/help/bounty).

***

No updates in ["Vue router 4 support" issue](https://github.com/nuxt-community/router-module/issues/110) of [@nuxtjs/router][1] since May 30, 2022 - looks like no way except dealing with manual routing myself if I need it.

I have checked the [source code][2] of **@nuxtjs/router** - the is no much code, but I have not understood where the manual routing has been set to Nuxt.

It is required to do 2 things:

1. Disable the default routing
2. Set the new routing

For the Nuxt 2.X, the disabling part was

```
if (!options.parsePages) {
  this.nuxt.hook('build:before', () => {
    this.nuxt.options.build.createRoutes = () => {
      return []
    }
  })
}
```

In the Nuxt 3.X case, there is no `createRoutes` property anymore.

Then, how I can to specify the new routing?
There is the dedicated property?

We can start with inline plugin definition:

```typescript
import { defineNuxtConfig, NuxtConfig } from "nuxt/config";


export default defineNuxtConfig({
  // ...
  modules: [
    async (inlineOptions: unknown, nuxt: NuxtConfig) => {

    }
  ]
});
```

By the way, if to try to use the current **@nuxtjs/router** for Nuxt 3, it will be [this error][3].

***

P. S. Please, no disputes "should I use the directory based routing?" in this topics.
If you are preferring the directory based routing - great, but I am not and please don't try to overpersuade mе.

  [1]: https://www.npmjs.com/package/@nuxtjs/router
  [2]: https://github.com/nuxt-community/router-module/blob/main/src/module.ts
  [3]: https://github.com/nuxt-community/router-module/issues/123

**Comment by TheAlexLichter** [comment, marked as answer]
Answered on SO with:

> You don't need the module anymore in Nuxt3. Now you can create a [`router.options.ts`](https://nuxt.com/docs/guide/directory-structure/pages#custom-routes) in the `/app` folder and edit or fully override the routes 😋

**Reply by TokugawaTakeshi** [reply]
Thank you for the answer! Works fine.
